### PR TITLE
Add MITRE ATT&CK Mappings to current SCBs

### LINKED ISSUE #937: Add MITRE ATT&CK TTP Mappings to M365 SCBs
# 💡 Summary #

Adding MITRE ATT&CK TTP Mapping to each baseline policy matching the formatting done in GWS's SCBs. 

## Motivation and context ##

The alignment of  the M365 SCB policies to their MITRE ATT&CK TTP mappings will allow the user of the SCBs to better understand the impact, specific threat models, and methodologies that are identified with our policies. 

The format of the mappings will also mirror the format of the GWS SCBs to ensure consistency between the two products.  

## Implementation notes ##

- Add in the TTP mappings for all SCBs and policies matching the formatting shown in the markdown style guide listed below
https://github.com/cisagov/ScubaGear/blob/Baseline-Markdown-Style-Guide/baselines/Example.md

## Acceptance criteria ##

How do we know when this work is done?

- [x] MITRE ATT&CK TTP mappings are added to the following SCBs
  - [x] AAD
  - [x] Defender
  - [x] EXO
  - [x] SharePoint
  - [x] Teams
  - [x] PowerBI
  - [x] PowerPlatform
- [x] All mappings reflect the formatting found in the markdown style guide

### PR DESCRIPTION
**Pull request by ahuynhMITRE**
## 🗣 Description ##

Added MITRE's ATT&CK TTP mappings and links to applicable security control baselines (SCBs) policies matching the format seen in GWS' SCBs

## 💭 Motivation and context ##

This update to the SCBs is required because it provides additional context into the tactics, techniques, and protocols the policies are attempting to harden against. This will also align the M365 SCBs with the GWS SCBs that currently have the mappings.

Closes #937 

## 🧪 Testing ##

- check if the formatting is consistent across all of the added TTP mappings
- check if the correct TTP mappings are aligned to each policy. Spreadsheet has been sent to @mitchelbaker-cisa for dissemination. 

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] Unit tests added/updated to cover PowerShell and Rego changes.
- [x] Functional tests added/updated to cover PowerShell and Rego changes.
- [x] All relevant functional tests passed.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [ ] PR passed smoke test check.
- [ ] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [ ] Resolved all merge conflicts on branch
- [ ] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: PowerShell/ScubaGear/baselines/teams.md [hunk 1]
```diff
@@ -604,13 +632,17 @@ Attachments included with Teams messages SHOULD be scanned for malware.
 <!--Policy: MS.TEAMS.7.1v1; Criticality: SHOULD -->
 - _Rationale:_ Teams can be used as a mechanism for delivering malware. In many cases, malware can be detected through scanning, reducing the risk for end users.
 - _Last modified:_ July 2023
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1566.001:Spearphishing Attachment](https://attack.mitre.org/techniques/T1566/001/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/teams.md:636]: Reference parent technique https://attack.mitre.org/techniques/T1566/

**Review by mitchelbaker-cisa — CHANGES_REQUESTED** [review]
Main formatting considerations: 

Bullets should start with the parent technique. If there are sub-techniques then list these as sub bullets for each baseline.

- [T1110: Brute Force](https://attack.mitre.org/techniques/T1110/)
   - [T1110.001: Password Guessing](https://attack.mitre.org/techniques/T1110/001/)
   - [T1110.002: Password Cracking](https://attack.mitre.org/techniques/T1110/002/)
   - [T1110.003: Password Spraying](https://attack.mitre.org/techniques/T1110/003/)

The techniques are formatted as [T1566:Phishing]. Let's add a space after the colon for all, [T1566: Phishing].

For consistency with GWS let's be sure to reference the parent technique for any sub-techniques.

**Comment by ahuynhMITRE** [comment]
thanks @mitchelbaker-cisa updated the mappings to do the following:

- add in reference parent ttp mapping
- format sub ttp mappings as sub bullets to these parent ttps
- added a space after the ":" for each policy

File: PowerShell/ScubaGear/baselines/aad.md [hunk 2]
```diff
@@ -502,20 +570,32 @@ A minimum of two users and a maximum of eight users SHALL be provisioned with th
...
 
 #### MS.AAD.7.3v1
 Privileged users SHALL be provisioned cloud-only accounts separate from an on-premises directory or other federated identity providers.
 
 <!--Policy: MS.AAD.7.3v1; Criticality: SHALL -->
 - _Rationale:_ By provisioning cloud-only Azure AD user accounts to privileged users, the risks associated with a compromise of on-premises federation infrastructure are reduced. It is more challenging for the adversary to pivot from the compromised environment to the cloud with privileged access.
 - _Last modified:_ June 2023
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1556: Modify Authentication Process](https://attack.mitre.org/techniques/T1556/)
+    - [T1556.007:Hybrid Identity](https://attack.mitre.org/techniques/T1556/007/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/aad.md:598]:
```suggestion
    - [T1556.007: Hybrid Identity](https://attack.mitre.org/techniques/T1556/007/)
```

File: PowerShell/ScubaGear/baselines/aad.md [hunk 3]
```diff
@@ -526,27 +606,38 @@ Permanent active role assignments SHALL NOT be allowed for highly privileged rol
...
+    - [T1098.003: Additional Cloud Roles](https://attack.mitre.org/techniques/T1098/003/)
 
 #### MS.AAD.7.5v1
 Provisioning users to highly privileged roles SHALL NOT occur outside of a PAM system.
 
 <!--Policy: MS.AAD.7.5v1; Criticality: SHALL -->
 - _Rationale:_ Provisioning users to privileged roles within a PAM system enables enforcement of numerous privileged access policies and monitoring. If privileged users are assigned directly to roles in the M365 admin center or via PowerShell outside of the context of a PAM system, a significant set of critical security capabilities are bypassed.
 - _Last modified:_ June 2023
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1651:Cloud Administration Command](https://attack.mitre.org/techniques/T1651/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/aad.md:620]:
```suggestion
  - [T1651: Cloud Administration Command](https://attack.mitre.org/techniques/T1651/)
```

File: PowerShell/ScubaGear/baselines/aad.md [hunk 4]
```diff
@@ -555,13 +646,22 @@ User activation of the Global Administrator role SHALL trigger an alert.
...
 #### MS.AAD.7.9v1
 User activation of other highly privileged roles SHOULD trigger an alert.
 
 <!--Policy: MS.AAD.7.9v1; Criticality: SHOULD -->
 - _Rationale:_ Closely monitor activation of high-risk roles for signs of compromise. Send activation alerts to enable the security monitoring team to detect compromise attempts. In some environments, activating privileged roles can generate a significant number of alerts.
 - _Last modified:_ June 2023
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1098: Account Manipulation](https://attack.mitre.org/techniques/T1098/)
+    - [T1098.003: Additional Cloud Roles](https://attack.mitre.org/techniques/T1098/003/)
+  - [T1136:Create Account](https://attack.mitre.org/techniques/T1136/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/aad.md:662]:
```suggestion
  - [T1136: Create Account](https://attack.mitre.org/techniques/T1136/)
```

Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/aad.md:662]:
```suggestion
  - [T1136: Create Account](https://attack.mitre.org/techniques/T1136/)
```

File: PowerShell/ScubaGear/baselines/aad.md [hunk 5]
```diff
@@ -555,13 +646,22 @@ User activation of the Global Administrator role SHALL trigger an alert.
...
+    - [T1136.003:Cloud Account](https://attack.mitre.org/techniques/T1136/003/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/aad.md:663]:
```suggestion
    - [T1136.003: Cloud Account](https://attack.mitre.org/techniques/T1136/003/)
```

File: PowerShell/ScubaGear/baselines/aad.md [hunk 6]
```diff
@@ -715,20 +815,31 @@ Guest users SHOULD have limited or restricted access to Azure AD directory objec
 <!--Policy: MS.AAD.8.1v1; Criticality: SHOULD -->
 - _Rationale:_ Limiting the amount of object information available to guest users in the tenant, reduces malicious reconnaissance exposure, should a guest account become compromised or be created by an adversary.
 - _Last modified:_ June 2023
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1087: Account Discovery](https://attack.mitre.org/techniques/T1087/)
+    - [T1087.003: Email Account](https://attack.mitre.org/techniques/T1087/003/)
+    - [T1087.004: Cloud Account](https://attack.mitre.org/techniques/T1087/004/)
+  - [T1526:Cloud Service Discovery](https://attack.mitre.org/techniques/T1526/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/aad.md:822]:
```suggestion
  - [T1526: Cloud Service Discovery](https://attack.mitre.org/techniques/T1526/)
```

File: PowerShell/ScubaGear/baselines/defender.md [hunk 7]
```diff
@@ -391,6 +436,10 @@ A custom policy SHALL be configured to protect PII and sensitive information, as
                provide a way for agencies to detect and prevent
                unauthorized disclosures.
 - _Last modified:_ June 2023
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1567 :Exfiltration Over Web Service](https://attack.mitre.org/techniques/T1567/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/defender.md:440]:
```suggestion
  - [T1567: Exfiltration Over Web Service](https://attack.mitre.org/techniques/T1567/)
```

File: PowerShell/ScubaGear/baselines/defender.md [hunk 8]
```diff
@@ -412,6 +467,10 @@ The action for the custom policy SHOULD be set to block sharing sensitive inform
 - _Last modified:_ June 2023
 - _Note:_ The custom policy referenced here is the same policy
           configured in [MS.DEFENDER.4.1v1](#msdefender41v1).
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1567:Exfiltration Over Web Service](https://attack.mitre.org/techniques/T1567/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/defender.md:471]:
```suggestion
  - [T1567: Exfiltration Over Web Service](https://attack.mitre.org/techniques/T1567/)
```

File: PowerShell/ScubaGear/baselines/defender.md [hunk 9]
```diff
@@ -412,6 +467,10 @@ The action for the custom policy SHOULD be set to block sharing sensitive inform
...
+  - [T1530:Data from Cloud Storage](https://attack.mitre.org/techniques/T1530/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/defender.md:472]:
```suggestion
  - [T1530: Data from Cloud Storage](https://attack.mitre.org/techniques/T1530/)
```

File: PowerShell/ScubaGear/baselines/defender.md [hunk 10]
```diff
@@ -412,6 +467,10 @@ The action for the custom policy SHOULD be set to block sharing sensitive inform
...
+  - [T1213:Data from Information Repositories](https://attack.mitre.org/techniques/T1213/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/defender.md:473]:
```suggestion
  - [T1213: Data from Information Repositories](https://attack.mitre.org/techniques/T1213/)
```

File: PowerShell/ScubaGear/baselines/exo.md [hunk 11]
```diff
@@ -915,7 +985,18 @@ Mailbox auditing SHALL be enabled.
 <!--Policy: MS.EXO.13.1v1; Criticality: SHALL -->
 - _Rationale:_ Exchange Online user accounts can be compromised or misused. Enabling mailbox auditing provides a valuable source of information to detect and respond to mailbox misuse.
 - _Last modified:_ June 2023
-
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1070: Indicator Removal](https://attack.mitre.org/techniques/T1070/)
+    - [T1070.008: Clear Mailbox Data](https://attack.mitre.org/techniques/T1070/008/)
+  - [T1098: Account Manipulation](https://attack.mitre.org/techniques/T1098/)
+    - [T1098.002:Additional Email Delegate Permissions](https://attack.mitre.org/techniques/T1098/002/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/exo.md:992]:
```suggestion
    - [T1098.002: Additional Email Delegate Permissions](https://attack.mitre.org/techniques/T1098/002/)
```

File: PowerShell/ScubaGear/baselines/powerbi.md [hunk 12]
```diff
@@ -389,6 +420,13 @@ Sensitivity labels SHOULD be enabled for Power BI and employed for sensitive dat
 <!--Policy: MS.POWERBI.7.1v1; Criticality: SHOULD -->
 - _Rationale:_ A document without sensitivity labels may be opened unknowingly, potentially exposing data to someone who is not supposed to have access to it. This policy will help organize and classify data, making it easier to keep data out of the wrong hands.
 - _Last modified:_ June 2023
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1048:Exfiltration Over Alternative Protocol](https://attack.mitre.org/techniques/T1048/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/powerbi.md:424]:
```suggestion
  - [T1048: Exfiltration Over Alternative Protocol](https://attack.mitre.org/techniques/T1048/)
```

File: PowerShell/ScubaGear/baselines/powerbi.md [hunk 13]
```diff
@@ -389,6 +420,13 @@ Sensitivity labels SHOULD be enabled for Power BI and employed for sensitive dat
...
+  - [T1567:Exfiltration Over Web Service](https://attack.mitre.org/techniques/T1567/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/powerbi.md:425]:
```suggestion
  - [T1567: Exfiltration Over Web Service](https://attack.mitre.org/techniques/T1567/)
```

File: PowerShell/ScubaGear/baselines/powerbi.md [hunk 14]
```diff
@@ -389,6 +420,13 @@ Sensitivity labels SHOULD be enabled for Power BI and employed for sensitive dat
...
+  - [T1213.002:Sharepoint](https://attack.mitre.org/techniques/T1213/002/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/powerbi.md:426]:
```suggestion
  - [T1213.002: Sharepoint](https://attack.mitre.org/techniques/T1213/002/)
```

File: PowerShell/ScubaGear/baselines/powerbi.md [hunk 15]
```diff
@@ -389,6 +420,13 @@ Sensitivity labels SHOULD be enabled for Power BI and employed for sensitive dat
...
+  - [T1213:Data from Information Repositories](https://attack.mitre.org/techniques/T1213/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/powerbi.md:427]:
```suggestion
  - [T1213: Data from Information Repositories](https://attack.mitre.org/techniques/T1213/)
```

File: PowerShell/ScubaGear/baselines/powerbi.md [hunk 16]
```diff
@@ -389,6 +420,13 @@ Sensitivity labels SHOULD be enabled for Power BI and employed for sensitive dat
...
+  - [T1530:Data from Cloud Storage](https://attack.mitre.org/techniques/T1530/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/powerbi.md:428]:
```suggestion
  - [T1530: Data from Cloud Storage](https://attack.mitre.org/techniques/T1530/)
```

File: PowerShell/ScubaGear/baselines/teams.md [hunk 17]
```diff
@@ -671,13 +711,27 @@ URL comparison with a blocklist SHOULD be enabled.
 <!--Policy: MS.TEAMS.8.1v1; Criticality: SHOULD -->
 - _Rationale:_ Users may be directed to malicious websites via links in Teams. Blocking access to known malicious URLs can help prevent users from accessing known malicious websites.
 - _Last modified:_ July 2023
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1204: User Execution](https://attack.mitre.org/techniques/T1204/)
+    - [T1204.001: Malicious Link](https://attack.mitre.org/techniques/T1204/001/)
+    - [T1204.002: Malicious File](https://attack.mitre.org/techniques/T1204/002/)
+  - [T1566: Phishing](https://attack.mitre.org/techniques/T1566/)
+    - [T1566.001: Spearphishing Attachment](https://attack.mitre.org/techniques/T1566/001/)
+  - [T1189:Drive-by Compromise](https://attack.mitre.org/techniques/T1189/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/teams.md:720]:
```suggestion
  - [T1189: Drive-by Compromise](https://attack.mitre.org/techniques/T1189/)
```

File: PowerShell/ScubaGear/baselines/teams.md [hunk 18]
```diff
@@ -671,13 +711,27 @@ URL comparison with a blocklist SHOULD be enabled.
...
 <!--Policy: MS.TEAMS.8.2v1; Criticality: SHOULD -->
 - _Rationale:_ Users may click on malicious links in Teams, leading to compromise or authorized data disclosure. Enabling user click tracking lets agencies know if a malicious link may have been visited after the fact to help tailor a response to a potential incident.
 - _Last modified:_ July 2023
+- _MITRE ATT&CK TTP Mapping:_
+  - [T1204: User Execution](https://attack.mitre.org/techniques/T1204/)
+    - [T1204.001: Malicious Link](https://attack.mitre.org/techniques/T1204/001/)
+    - [T1204.002: Malicious File](https://attack.mitre.org/techniques/T1204/002/)
+  - [T1566: Phishing](https://attack.mitre.org/techniques/T1566/)
+    - [T1566.001: Spearphishing Attachment](https://attack.mitre.org/techniques/T1566/001/)
+  - [T1189:Drive-by Compromise](https://attack.mitre.org/techniques/T1189/)
```
Comment (mitchelbaker-cisa) [inline PowerShell/ScubaGear/baselines/teams.md:734]:
```suggestion
  - [T1189: Drive-by Compromise](https://attack.mitre.org/techniques/T1189/)
```

**Review by mitchelbaker-cisa — APPROVED** [review]
Thanks Andrew, looks good!

**Review by rmoffitt-m — APPROVED** [review]

**Comment by ahuynhMITRE** [comment]
@nanda-katikaneni good to merge!

**Pull request merged**